lib/jwt/jwt.go

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jwt
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+	josejwt "gopkg.in/square/go-jose.v2/jwt"
+)
+
+// Config defines the clock and PEM encoded bytes of a public and private
+// key that form a *jwt.Key.
+type Config struct {
+	// Clock is used to control expiry time.
+	Clock clockwork.Clock
+
+	// PublicKey is a PEM encoded key that can be used to verify a signed token.
+	PublicKey []byte
+
+	// PrivateKey is a PEM encoded key that can be used to sign and verify tokens.
+	PrivateKey []byte
```

Comment:
Pass around `crypto.Signer` and `crypto.PublicKey`.
Encoding/decoding of key pairs shouldn't be the responsibility of this package. Plus this allows us to use external private keys (e.g. HSM)

I will skip similar comments in the code below, but you should similarly avoid hardcoding `rsa.*Key`